Content.Client/UserInterface/Systems/Chat/Widgets/ChatBox.xaml.cs

```diff
@@ -95,12 +174,22 @@ private void OnChannelFilter(ChatChannel channel, bool active)
         }
     }
 
-    public void AddLine(string message, Color color)
+    public void AddLine(string message, Color color, int repeat = 0) // EE - Chat stacking - repeat
     {
-        var formatted = new FormattedMessage(3);
+        var formatted = new FormattedMessage(4);
         formatted.PushColor(color);
         formatted.AddMarkupOrThrow(message);
         formatted.Pop();
+        // EE - Chat stacking
+        if (repeat != 0)
+        {
+            int displayRepeat = repeat + 1;
+            int sizeIncrease = Math.Min(displayRepeat / 6, 5);
+            formatted.AddMarkup(_loc.GetString("chat-system-repeated-message-counter",
+                                ("count", displayRepeat),
+                                ("size", 8 + sizeIncrease)
+                                ));
```

Comment:
'FormattedMessage.AddMarkup(string)' is obsolete: 'Use AddMarkupOrThrow or TryAddMarkup'
```suggestion
            formatted.AddMarkupOrThrow(_loc.GetString("chat-system-repeated-message-counter",
                                ("count", displayRepeat),
                                ("size", 8 + sizeIncrease)
                                ));
```